app/models/book.rb

```diff
@@ -1,6 +1,35 @@
 class Book < ApplicationRecord
+  BOOK_ATTRS = %i(name description price page_num publisher_name
+    quantity category_id author_id image).freeze
+
   has_many :order_details, dependent: :nullify
   has_many :comments, dependent: :destroy
+  has_one_attached :image
   belongs_to :author
   belongs_to :category
+
+  validates :name, presence: true,
+            length: {maximum: Settings.max_name_book_length}
+  validates :price, presence: true,
+            numericality: {greater_than: Settings.min_book_price_length,
+                           less_than: Settings.max_book_price_length}
+  validates :description, presence: true,
+            length: {maximum: Settings.max_description_length}
+  validates :image,
+            content_type: {in: Settings.image_format,
+                           message: I18n.t(".invalid_format")},
```

Comment:
e dùng :invalid_format
xong định nghĩa error message theo chuẩn trong file i18n là sẽ dc
errors:
  ten_model:
....

Reply:
tham khảo
https://stackoverflow.com/questions/17935597/ruby-on-rails-i18n-want-to-translate-custom-messages-in-models